share/nisar/examples/noise_spectra_analysis.py

```diff
@@ -0,0 +1,450 @@
+                    f"Threshold used to classify power spectral density values as contaminated "
+                    f"by radio frequency interference"
+                )
+                dset_thresh.attrs["units"] = np.bytes_("decibel re 1/hertz")
+
+                # Initialize hit count array
+                rfi_bin_hit_count = np.zeros(num_freq_bins, dtype=int)
+                rfi_bin_max_streak = np.zeros(num_freq_bins, dtype=int) 
+                
+                # Compute averaged power spectra density and RFI hit count
+                for az_idx, az_slice in enumerate(slices):
+                    az_blk = raw_data[az_slice, :]  # Shape: (num_pulses_blk, range_samples)
+                    _, psd_az_blk = welch(
+                        az_blk, 
+                        fs=fs, 
+                        nperseg=num_fft, 
+                        nfft=num_fft, 
+                        detrend=False, 
+                        axis=1, 
+                        return_onesided=False
+                    )
+                    psd_az_blk_db = 10*np.log10(fftshift(psd_az_blk))
+
+                    # Compare each frequency bin in each pulse against the threshold
+                    rfi_mask_blk = psd_az_blk_db > threshold_db_hz  # shape (num_pulses_blk, num_fft)
+
+                    # Update RFI stats: Collect RFI hit data, and RFI hit max streak data
+                    update_rfi_hit_stats(rfi_mask_blk, rfi_bin_hit_count, rfi_bin_max_streak)
+
+                    # Compute Average range frequency power spectra density (dB/Hz)
+                    avg_psd_az_blk = np.mean(fftshift(psd_az_blk), axis=0)
+                    avg_psd_az_blk_db = 10 * np.log10(avg_psd_az_blk)
+
+                    # Write PSD of each block
+                    dset_psd[az_idx, :] = avg_psd_az_blk_db
+
+                # Write RFI Hit Count
+                if "interferenceHitCount" in out_grp:
+                    del out_grp["interferenceHitCount"]
+                dset_hit = out_grp.create_dataset('interferenceHitCount', data=rfi_bin_hit_count)
+                dset_hit.attrs["description"] =  np.bytes_(
+                    f"Count of pulses where RFI detection threshold was exceeded "
+                    f"(in each frequency bin)"
+                )
+                dset_hit.attrs["units"] = np.bytes_("1")
+
+                # Write RFI Hit Max Streak
+                if "interferenceMaxStreak" in out_grp:
+                    del out_grp["interferenceMaxStreak"]
+                dset_streak = out_grp.create_dataset('interferenceMaxStreak', data=rfi_bin_max_streak)
+                dset_streak.attrs["description"] =  np.bytes_(
+                    f"Maximum number of consecutive pulses where RFI detection threshold was exceeded "
+                    f"(in each freuqency bin)"
+                )
+                dset_streak.attrs["units"] = np.bytes_("1")
+
+                print(f'Processed: frequency{freq} {pol}')
+                print()
```

Comment:
Just a minor suggestion:
```suggestion
                print(f'Processed: frequency{freq} {pol}\n')
```